### ISSUE TITLE
Cannot use "mixed" as class name

### ISSUE THREAD

**Issue by lindelius** [mockery/mockery#1149]
We're getting the following error when running our PHPUnit suite:
> Cannot use 'mixed' as class name as it is reserved in /tmp/MockerykqgHQz on line 1

PHP version: 8.0.8
Mockery version: 1.4.4
PHPUnit version: 9.5.10

---

The contents of the generated file:
```php
<?php class mixed {} 
```

**Comment by ghostwriter** [comment]
Attempting to declare a class with name `mixed` in [PHP 8: mixed-type](https://php.watch/versions/8.0/mixed-type) results the following error:
`Fatal error: Cannot use 'mixed' as class name as it is reserved in ... on line …` https://3v4l.org/hebJi

do you have a test, that attempts to mock a class with the name `mixed`?

**Comment by lindelius** [comment]
Hi,

The issue is apparently caused by having methods with a return type of `mixed`, and then when mocking that class adding `shouldReceive` that then expects the method to return an object. It's resolved by simply removing the `mixed` type hint.

```php
interface MyInterface {
    public function getSomething(): mixed;
}

$object = Mockery::mock(MyInterface::class);
$object->shouldReceive('getSomething->someMethod')->andReturn("someValue");
```

Seems like Mockery should know that `mixed` is a reserved keyword and just treat it as a `stdClass` object or whatever. The same code works fine without the return type on the method.

**Comment by ramsey** [comment]
I have a reproducible example in #1155, which is the same type of bug with the `static` return type.

**Comment by ghostwriter** [comment]
Hey @lindelius and @ramsey,

your respective issues have been resolved via #1156 and #1157 (awaiting code review cc: @davedevelopment @padraic).
